### PR TITLE
#INS-107- Data Model for Basic observation 

### PR DESCRIPTION
**Pull request by fwpushan**
1. Add all code tables for basic observation
2. Update Endpoint API
3. Update tests

### CODE REVIEW

File: api/api_sources/sources/database/database-schema/schema-sqls/SoilTextureCodeSchema.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ALTER TABLE soil_texture_code ADD COLUMN soil_texture_code VARCHAR(1) NOT NULL UNIQUE;
+ALTER TABLE soil_texture_code ADD COLUMN description VARCHAR(100) NULL;
+ALTER TABLE soil_texture_code ADD COLUMN active_ind BOOLEAN NOT NULL DEFAULT TRUE;
+
+
+        
+-- ### Creating Comments on table ### --
+
+        
+COMMENT ON TABLE soil_texture_code IS 'Code table for schema of soil texture codes';
```
Comment (klefler) [inline api/api_sources/sources/database/database-schema/schema-sqls/SoilTextureCodeSchema.sql]: I'd like to recommend we add a more descriptive table comment that describes "soil texture code" from a business perspective.

For example (from IAPP); The soil texture code provides details regarding the quality and water-retention of soil in a site where an invasive plant has been identified. Values are C (Coarse), M (Medium), F (Fine), O (Organic).

File: api/api_sources/sources/database/database-schema/schema-sqls/SpecificUseCodeSchema.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ALTER TABLE specific_use_code ADD COLUMN specific_use_code VARCHAR(2) NOT NULL UNIQUE;
+ALTER TABLE specific_use_code ADD COLUMN description VARCHAR(100) NULL;
+ALTER TABLE specific_use_code ADD COLUMN active_ind BOOLEAN NOT NULL DEFAULT TRUE;
+
+
+        
+-- ### Creating Comments on table ### --
+
+        
+COMMENT ON TABLE specific_use_code IS 'Code table for schema of Specific use codes of the observation';
```
Comment (klefler) [inline api/api_sources/sources/database/database-schema/schema-sqls/SpecificUseCodeSchema.sql]: I'd like to recommend we add a more descriptive table comment that describes "specific use code" from a business perspective.

For example; The specific use code provides details regarding the use of an area where an observation has been made. This assists in determining the appropriate response and treatment methods when dealing with the observation. Values are GR (Gravel pit), NS (No-spray buffer), NH (Numbered Highway), PL (Parking Lot), RT (Rec Site/Trail), RA (Rest Area), SS (Sensitive Site), TL (Transfer station/Landfil), WP (Within PFZ), YW (Yard/Ditching waste dump).

File: api/api_sources/sources/database/database-schema/schema-sqls/SurveyGeometryCodeSchema.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ALTER TABLE survey_geometry_code ADD COLUMN survey_geometry_code VARCHAR(2) NOT NULL UNIQUE;
+ALTER TABLE survey_geometry_code ADD COLUMN description VARCHAR(100) NULL;
+ALTER TABLE survey_geometry_code ADD COLUMN active_ind BOOLEAN NOT NULL DEFAULT TRUE;
+
+
+        
+-- ### Creating Comments on table ### --
+
+        
+COMMENT ON TABLE survey_geometry_code IS 'Code table for schema of survey geometry codes';
```
Comment (klefler) [inline api/api_sources/sources/database/database-schema/schema-sqls/SurveyGeometryCodeSchema.sql]: I'd like to recommend we add a more descriptive table comment that describes "survey geometry code" from a business perspective.

For example; Describes a 2 or 3 dimensional geometry used to describe the spatial extent of an observation: Point, linear corridor, transect, circle, rectangular plot, stratified grid, radio tract, GPS tract, water body outline.

File: api/api_sources/sources/database/database-schema/schema-sqls/SpecificUseCodeSchema.sql (hunk 2 above)
Comment (fwpushan) [reply api/api_sources/sources/database/database-schema/schema-sqls/SpecificUseCodeSchema.sql]: Thanks! 
I will update SQL file

**Pull request merged**